fedimint/src/lib.rs

```diff
@@ -145,6 +177,102 @@ impl FedimintServer {
         }
     }
 
+    // Build a `ConsensusOutcome` then use the API to validate and process missed epochs
+    pub async fn rejoin_consensus(&mut self) {
+        let mut msg_buffer = vec![];
+        let next_epoch_num = self.determine_rejoin_epoch(&mut msg_buffer).await;
+        tracing::info!("Rejoining consensus: at epoch {}", next_epoch_num);
+        self.hbbft.skip_to_epoch(next_epoch_num);
+
+        let mut outcomes: Vec<ConsensusOutcome> = vec![];
+        for msg in msg_buffer {
+            outcomes.append(&mut self.handle_message(msg).await);
+        }
+        while outcomes.is_empty() {
+            let msg = self.connections.receive().await;
+            outcomes = self.handle_message(msg).await;
+        }
+        tracing::info!("Rejoining consensus: created outcome");
+        // FIXME: Should handle failing and querying other peers
+        self.download_history(outcomes[0].clone()).await.unwrap();
```

Comment:
Actually it's possible that we create epochs from the near-future.  This was another possible solution to the resync I tried to explain on our call, but I prefer the custom messages solution in this PR because it'll be more useful when I add the handling of a consensus restart scenario.

This is actually an assumption that occurs in our normal hbbft handling code (that epochs are received in-order) and should be tracked as a separate issue (I'll create one for that now).